bin/github-comment-manager.js

```diff
@@ -0,0 +1,70 @@
+const request = require('request');
+
+const existingComments = (accountName, accountKey, issueNumber) =>
+  new Promise((resolve, reject) => {
+    const auth = 'Basic ' + new Buffer(accountName + ':' + accountKey).toString('base64');
+    const options = {
+      url: `https://api.github.com/repos/newsuk/times-components/issues/${issueNumber}/comments`,
+      headers: {
+       Authorization: auth,
+       'User-Agent': accountName
+      }
+    };
+
+    request.get(options, (error, response, body) => {
+      if (error) reject(error);
+      const ids = JSON.parse(body)
+        .filter(({ user }) => user.login === accountName)
+        .map(({ id }) => id);
+      
+      resolve(ids);
+    });
+});
+
+const deleteComments = (commentsToDelete, accountName, accountKey) =>
+  new Promise((resolve, reject) => {
+    commentsToDelete.forEach((commentId) => {
+      const auth = 'Basic ' + new Buffer(accountName + ':' + accountKey).toString('base64');
+      const deleteOptions = {
+        url: `https://api.github.com/repos/newsuk/times-components/issues/comments/${commentId}`,
+        headers: {
+          Authorization: auth,
+          'User-Agent': accountName
+        }
+      };
+
+      request.delete(deleteOptions, (error) => {
+        if (error) reject(error);
+      });
+    });
+    resolve('deleted');
```

Comment:
This is still not right, this calls resolve even before any requests are finished